src/coreclr/vm/jitinterface.cpp

```diff
@@ -10722,12 +10733,15 @@ void* CEEJitInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,         /* IN  */
                     {
                         // Cache it for future uses to avoid taking the lock again.
                         hlpFinalTierAddrTable[dynamicFtnNum] = finalTierAddr;
+                        EE_TO_JIT_TRANSITION_LEAF();
```

Comment:
This should use `goto exit` pattern used elsewhere and/or factored into a helper method. These macros are not meant to be used multiple times per method like this. It happens to work for the LEAF macro, but it would not work for the non-LEAF macro.